cmd/extensions-validator/cmd/validate.go

```diff
@@ -0,0 +1,44 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/siderolabs/talos/pkg/machinery/extensions"
+	"github.com/spf13/cobra"
+)
+
+var validateCmd = &cobra.Command{
+	Use:   "validate",
+	Short: "Validate the extensions rootfs",
+	Long:  `Usage: extensions-validator validate`,
+	// define a rootfs path argument
+	Args: cobra.NoArgs,
+	RunE: func(_ *cobra.Command, _ []string) error {
+		return validateRootfs()
+	},
+}
+
+var rootfsPath string
+
+func init() {
+	validateCmd.Flags().StringVar(&rootfsPath, "rootfs", "", "Path to the rootfs")
+	rootCmd.AddCommand(validateCmd)
+}
+
+func validateRootfs() error {
+	if rootfsPath == "" {
+		return errors.New("rootfs path is required")
+	}
+
+	extension, err := extensions.Load(rootfsPath)
+	if err != nil {
+		return fmt.Errorf("error loading extension: %w", err)
+	}
+
+	return extension.Validate()
```

Comment:
it'd be cool to add an option to validate that extension name in the manifest matches expected (passed via the flag)

also we should validate (probably should first update Talos) that extension version in the manifest is valid (whatever valid we define, but e.g. it should not accept `$VERSION` as the version)

Reply:
fixed